src/Tribe/Attendee_Registration/Modal.php

```diff
@@ -68,24 +86,41 @@ function modal_cart_template( $content, $template_obj ) {
 	 * @param Tribe__Tickets__Editor__Template $template_obj the Template object.
 	 */
 	function append_modal_ar_template( $unused_content, $template_obj ) {
-		$template = 'modal/registration-js.php';
-		$file     = $this->locate_template( $template );
+		// If they're not using the new views, include v1 and bail.
+		if ( ! tribe_tickets_new_views_is_enabled() ) {
+			$this->append_modal_ar_template_v1( $unused_content, $template_obj );
+			return;
+		}
+
+		$obj_tickets  = $template_obj->get( 'tickets', [] );
+		$tickets_data = [];
 
-		$obj_tickets = $template_obj->get( 'tickets', [] );
-		$tickets = [];
 		foreach ( $obj_tickets as $ticket ) {
-			$ticket_data = array(
+			$tickets_data[] = [
 				'id'       => $ticket->ID,
 				'qty'      => 1,
 				'provider' => $ticket->get_provider(),
-			);
-
-			$tickets[] = $ticket_data;
+			];
 		}
 
+		/** @var Tribe__Tickets__Attendee_Registration__View $view */
+		$view = tribe( 'tickets.attendee_registration.view' );
+
+		$args = [
+			'tickets'        => $obj_tickets,
+			'providers'      => wp_list_pluck( $tickets_data, 'provider' ),
+			'provider_class' => '',
+			'has_tpp'        => false,
+			'post_id'        => $template_obj->get( 'post_id' ),
+			'view'           => $view,
+			'is_modal'       => true,
+		];
+
+		$template_obj->add_template_globals( $args );
+
 		ob_start();
 
-		include $file;
+		$template_obj->template( 'v2/modal/attendee-registration' );
 
 		$content = ob_get_clean();
```

Comment:
We should remove the output buffering because it's just echoing anyways. I have no idea why it was even this way before..